packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1299,6 +1309,69 @@ export class KeyringController extends BaseController<
     return seedWords;
   }
 
+  /**
+   * Select a keyring and execute the given function with
+   * the selected keyring, as a mutually exclusive atomic
+   * operation.
+   *
+   * The method automatically persists changes at the end of the
+   * function execution, or rolls back the changes if an error
+   * is thrown.
+   *
+   * @param selector - Keyring selector object.
+   * @param fn - Function to execute with the selected keyring.
+   * @param options - Additional options.
+   * @param options.createIfMissing - Whether to create a new keyring if the selected one is missing.
+   * @param options.createWithData - Optional data to use when creating a new keyring.
+   * @returns Promise resolving to the result of the function execution.
+   */
+  async withKeyring<K extends EthKeyring<Json> = EthKeyring<Json>, T = void>(
```

Comment:
Note that single-letter type parameters are disallowed by the latest version of our ESLint config (see https://github.com/MetaMask/eslint-config/pull/292). Is there a more descriptive name we could use here?

It would also be great to have `@template` entries for each one in the TSDoc comment, to explain what they represent